src/hardware/hantek-6xxx/api.c

```diff
@@ -551,8 +551,14 @@ static uint32_t data_amount(const struct sr_dev_inst *sdi)
 		data_left = devc->samplerate * NUM_CHANNELS;
 	}
 
+#ifdef __APPLE__
+	uint32_t i = 512;
+	while (i < data_left)
+		i *= 2;
+	data_left = i;
+#else
 	data_left += MIN_PACKET_SIZE; /* Driver does not handle small buffers. */
-
+#endif
```

Comment:
Linux add support for all kind of broken stuff, so this might be one of those quirks. Apple doesn't seem to care about random out of spec usb devices. Anyway in this case the proper fix it to adjust the driver.